src/Microsoft.Framework.Localization/ResourceManagerStringLocalizer.cs

```diff
@@ -185,43 +184,34 @@ private IEnumerable<string> GetResourceNamesFromCultureHierarchy(CultureInfo sta
             return resourceNames;
         }
 
-        private class MissingManifestCacheKey : IEquatable<MissingManifestCacheKey>
+        private IList<string> GetResourceNamesForCulture(CultureInfo culture)
         {
-            private readonly int _hashCode;
-
-            public MissingManifestCacheKey(string name, CultureInfo culture)
+            var resourceStreamName = ResourceBaseName;
+            if (!string.IsNullOrEmpty(culture.Name))
             {
-                Name = name;
-                CultureInfo = culture;
-                _hashCode = new { Name, CultureInfo }.GetHashCode();
+                resourceStreamName += "." + culture.Name;
             }
+            resourceStreamName += ".resources";
 
-            public string Name { get; }
+            var cacheKey = $"assembly={ResourceAssembly.FullName};resourceStreamName={resourceStreamName}";
 
-            public CultureInfo CultureInfo { get; }
-
-            public bool Equals(MissingManifestCacheKey other)
+            var cultureResourceNames = _resourceNamesCache.GetOrAdd(cacheKey, key =>
             {
-                return string.Equals(Name, other.Name, StringComparison.Ordinal)
-                    && CultureInfo == other.CultureInfo;
-            }
-
-            public override bool Equals(object obj)
-            {
-                var other = obj as MissingManifestCacheKey;
-
-                if (other != null)
+                var names = new List<string>();
+                using (var cultureResourceStream = ResourceAssembly.GetManifestResourceStream(key))
+                using (var resources = new ResourceReader(cultureResourceStream))
                 {
-                    return Equals(other);
+                    foreach (DictionaryEntry entry in resources)
```

Comment:
`var` doesn't work here because it's not `IEnumerable<T>`, just plain old `IEnumerable`. Yay old APIs! :smile: